dkg/pedersen/pedersen.go

```diff
@@ -488,6 +491,230 @@ func (w worker) work(jobIndex int) error {
 	return nil
 }
 
+// EncryptSecret implements dkg.Actor.
+func (a *Actor) EncryptSecret(msg []byte) (U kyber.Point, Cs []kyber.Point) {
+	pubK, err := a.GetPublicKey()
+	if err != nil {
+		dela.Logger.Error().Msgf("Cannot encrypt secret: %v", err.Error())
+	}
+
+	r := suite.Scalar().Pick(suite.RandomStream())
+	C := suite.Point().Mul(r, pubK)
+	dela.Logger.Debug().Msgf("C:%v", C)
+
+	U = suite.Point().Mul(r, nil)
+	dela.Logger.Debug().Msgf("U is:%v", U.String())
+
+	for len(msg) > 0 {
+		kp := suite.Point().Embed(msg, suite.RandomStream())
+		dela.Logger.Debug().Msgf("Keypoint:%v", kp.String())
+		dela.Logger.Debug().Msgf("X:%v", pubK.String())
+
+		Cs = append(Cs, suite.Point().Add(C, kp))
+		dela.Logger.Debug().Msgf("Cs:%v", C)
+
+		msg = msg[min(len(msg), kp.EmbedLen()):]
+	}
+
+	return
+}
+
+type OCS struct {
+	U    kyber.Point // U is the random part of the encrypted secret
+	pubk kyber.Point // The client's public key
+
+	nbnodes    int // How many nodes participate in the distributed operations
+	nbfailures int // How many failures occurred so far
+	threshold  int // How many replies are needed to re-create the secret
+
+	replies []types.ReencryptReply // replies received
+	Uis     []*share.PubShare      // re-encrypted shares
+}
+
+// newOCS creates a new on-chain secret structure.
+func newOCS(pubk kyber.Point) *OCS {
+	return &OCS{
+		pubk: pubk,
+	}
+}
```

Comment:
should the OCS-related logic be moved to its own file, for readability ?

Reply:
done